### PR TITLE
[Task 1871987] migrate the code to create redis from track1 to track2

### PR DESCRIPTION
**Pull request by andxu**
[AB#1871987](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1871987)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/createrediscache/CreateRedisCacheForm.java [hunk 1]
```diff
@@ -393,24 +375,21 @@ public void widgetSelected(SelectionEvent e) {
...
-                                }
-                            });
-                        } catch (Exception ex) {
-                            LOG.log(MessageHandler.getCommonStr(LOAD_LOCATION_AND_RESOURCE_ERROR), ex);
-                        }
+        AzureTaskManager.getInstance().runInBackground(
+                MessageHandler.getResourceString(resourceBundle, LOADING_LOCATION_AND_GRPS), false,
+                () -> {
+                    try {
+                        DefaultLoader.getIdeHelper().invokeLater(new Runnable() {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/forms/createrediscache/CreateRedisCacheForm.java:380]: AzureTaskManager

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/store/EclipseStore.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    		node.remove(combineKey(service, key));
+    		return;
+    		
+    	}
+        node.put(combineKey(service, key), value);
+        try {
+			node.flush();
+		} catch (BackingStoreException e) {
+            throw new AzureToolkitRuntimeException(e.getMessage(), e);
+		}
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/store/EclipseStore.java:46]: code style.

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/store/EclipseSecureStore.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class EclipseSecureStore implements ISecureStore {
+    ISecurePreferences node;
+
+    public EclipseSecureStore() {
+        ISecurePreferences preferences = SecurePreferencesFactory.getDefault();
+        node = preferences.node("com.microsoft.azuretools.core");
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/store/EclipseSecureStore.java:26]: why not use Plugin id as node key?

**Review by Flanker32 — APPROVED** [review]
Please resolve the comments in later PR

File: PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/store/EclipseStore.java (hunk 2 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/store/EclipseStore.java:46]: will fix in new PR

**Pull request merged**